demo/client/backend/mock/server.go

```diff
@@ -327,20 +351,24 @@ func query(c *gin.Context) {
 		panic("stub is nil!")
 	}
 
-	stub.MockStub.ChannelID = channelName
-	user := c.GetHeader("x-user")
-	stub.Creator = user
-
+	// prepare arguments
 	args, err := parsePayload(c)
 	if err != nil {
 		logger.Error(fmt.Sprintf("Request Error: %s\n", err))
 		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
+	user := c.GetHeader("x-user")
 
 	logger.Debugf("%s queries %s", user, args)
+
+	// invoke transactions
```

Comment:
better also add the RLock here?

Reply:
no, you also need a write lock here: you are locking not only the state inside the mock object but also the mock-wrapper object which has to be protected ..

Reply:
Oops, now i see what you mean: fixed now ...